style_guide.md

```diff
@@ -0,0 +1,217 @@
+
+- Shared blocks
+
+    - Blocks from `individual` folder are used in both household and individual schema.
+
+- Top level Jsonnet files
+
+    - For each schema there are lists of block and placeholder imports.
+    - Schema structure is divided into sections, named after given schema and ending with `.jsonnet` suffix.
+
+- Lib folders
+
+    - Rules and placeholders are located in `lib` folder in corresponding schemas along with `blocks` folder.
+
+- Rules and placeholders
+
+    - These are separated into: `common_rules`, `placeholders` and `rules`. These files should end with `.libsonnet` suffix.
+
+- Top level and extended
+
+    - `common_rules` and `placeholders` are located in top-level "lib" folder.
+    - `rules` files for each schema are located in schema's sub-folder and extend some of the top-level `common_rules`.
+
+## Global variables
+
+- Top level arguments
+
+    - For each schema created it is possible to pass variables, e.g.: `region_code` and `census_month_year_date`, with different outputs being generated depending on their value.
+    - These are defined in eq-questionnaire-runner, are known at questionnaire launch and won't change during the Census.
+
+- External variables
+
+    - External variables are being used. Example of that is `census_month_year_date`.
+
+- Region Code
+
+    - Depending on `region_code` variable's value, different structure of sections is generated (e.g. certain questions omitted).
+
+## Blocks
+
+### Structure
+
+#### Block json
+- There are two distinct styles we use with block json:
+  - Simple (no variants):
+    ```
+    // Imports
+    
+    {
+      // Block JSON
+    }
+    ```
+  - Variants:
+    ```
+    // Imports
+    
+    // Variant helper methods
+    local questionTitle(variantParam) = (
+      // Variables relevant to this method
+      // Return relevant string or JSON object
+    );
+    
+    // Main variant method - 'question' or 'content'
+    local question(variantParam) = {
+      // Variant JSON
+    };
+    
+    {
+      // Block JSON
+    }
+    ```
+- Question should be called with question(isProxy=true) or question(isProxy=false):
+```
+{
+  type: 'Question',
+  id: 'confirm_dob',
+  question_variants: [
+    {
+      question: question(isProxy=false)
+    },
+    {
+      question: question(isProxy=true)
+    },
+  ],
+}
+```
+- In the question method, when we need to resolve title, we call a new questionTitle method, passing isProxy:
+```
+local question(isProxy) = {
+  title: questionTitle(isProxy)
+  id: 'confirm-date-of-birth',
+  title: title,
+  type: 'General',
+  answers: [
+  {
+    id: 'confirm-date-of-birth-answer',
+    mandatory: true,
+    options: [
+      {
+        label: 'Yes',
+        value: 'Yes',
+       },
+      {
+        label: 'No',
+        value: 'No',
+      },
+    ],
+    type: 'Radio',
+    },
+  ],
+};
+
+```
+- In the questionTitle method we would return either of the two strings, based on the isProxy variable:
```

Comment:
```suggestion
- In the questionTitle method we would return either of the two strings, based on the `isProxy` variable:
```

Reply:
Some instances of words e.g.`isProxy` have been highlighted where others have not, would be good to double check through the doc and make it consistent throughout.